fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java

```diff
@@ -503,17 +506,23 @@ private SumoVehicleState processVehicleSubscriptionResult(final long time,
         final boolean isWaitingToLeaveParking = !isNewVehicle && hasInvalidPosition // check for "waiting" state
                 && sumoVehicle.lastVehicleData.getVehicleStopMode().isParking(); // check if was parked
 
-        if (hasInvalidPosition && isNewVehicle) {
-            /* if a vehicle has not yet been simulated but loaded by SUMO, the vehicle's position will be invalid.
-             * We ignore this vehicle until it's added to the simulation. */
-            log.debug("Skip vehicle {} which is loaded but not yet simulated.", veh.id);
-            return null;
-        }
-
         if (hasInvalidPosition && !isWaitingToLeaveParking) {
-            /* If the vehicle, however,  has already been in the simulation (lastVehicleData was valid),
-             * then there seems to be an error and the vehicle state will not be updated, resulting in a removal of the vehicle. */
-            log.warn("vehicle {} has no valid position and will be removed.", veh.id);
+            if (isNewVehicle) {
+                /* If a vehicle has not yet been simulated but loaded by SUMO, the vehicle's position will be invalid.
+                 * We ignore this vehicle until it's added to the simulation. */
+                log.debug("Skip vehicle {} which is loaded but not yet simulated.", veh.id);
+                return null;
+            }
+            boolean isTeleporting = getTeleportingList.execute(bridge).contains(veh.id); // we check this here to limit SUMO API calls
```

Comment:
Ah, I like that you call this only when necessary :) I guess we can neglect that we would execute this call multiple times if we have more than one teleporting vehicle?

Reply:
Yeah, it took a while to figure out the behavior of SUMO regarding teleporting vehicles.

I added a caching list, which is reset after each time step